jira/jira-v1.asciidoc

```diff
@@ -0,0 +1,104 @@
+* Stop Condition Key: Reference the “key” in the response, that needs to be met in order to stop the pagination. To paginate issues, it must be “issues”, to paginate projects it must be 'values'
+* Stop Condition Value: Reference the “value” in the response, that needs to be met in order to stop the pagination. For the jira config, to stop pagination the list of objects retrieved must be empty, then the stop condition should be []
+
+
+== Variables used
+
+The Jira REST configuration variables used are:
+
+* `${LW_BATCH_SIZE}` - This variable is used to set the `limit` query parameter, which controls the number of results that are returned in the response for both `blogpost` and `pages`.
+
+* `${LW_INDEX_START}` - Used with pagination feature. This variable is used to set the startAt query parameter, which is used to traverse the pagination. Jira pagination is zero-index-based. This means the first page number is 0.
+
+* `${LW_PARENT_DATA_KEY}` - This variable is used to access a value of the response from the `main request` to use in an `additional request`. The jira use case indicates this variable can be added the URL to execute a GET request to retrieve comments for blogposts and pages.
+
+
+== Endpoints to configure with the Jira REST connector
+
+The following table describes the jira REST connector endpoints.
+
+*Notes:*
+
+The list of objects is retrieved in batches. Objects include spaces, pages, blogs, and comments. The default number of batch items retrieved is 25. You can configure the query limit to retrieve more objects. The maximum value to retrieve is 250.
+
+
+[options="header",cols="1m,1,1m,1,1"]
+|=======================
+|Request type|Endpoint|HTTP operation |Query parameter |Description 
+
+|Root Request |/rest/api/2/search|GET    |fields=assignee,issuetype,priority,project,reporter,status,summary,updated,attachment&startAt=1&maxResults=50|Returns all jira issues. Each issue contains its list of attachments metadata
+|Child Request |/rest/api/2/issue/${LW_PARENT_DATA_KEY}/comment|GET|startAt=0&maxResult=50|Returns all comments. The request requires the 'issue Id'. Internally, the plugin replaces the variable ${LW_PARENT_DATA_KEY} with the 'issue id', which value is extracted from the 'issue object' (root object) by setting the parentDataKey property.
+|Child Request |/rest/api/2/issue/${LW_PARENT_DATA_KEY}/worklog | GET |startAt=0&maxResult=50 |Returns all worklogs. The request requires the 'issue Id'. Internally, the plugin replaces the variable ${LW_PARENT_DATA_KEY} with the 'issue id', which value is extracted from the 'issue object' (root object) by setting the parentDataKey property.
+|Child Request |/rest/api/2/attachment/content/${LW_PARENT_DATA_KEY}|GET|-|Use root response containing attachment Id's to retrieve attachments. The following jsonpath is used to retrieve the attachment id's `fields.attachment[*]` 
```

Comment:
@DavidDwyer87 I suggest to change description to: Download the attachment content. The request requires the 'attachment Id'. Internally, the plugin replaces the variable ${LW_PARENT_DATA_KEY} with the 'attachment id', which value is extracted in two steps: 1) setting prop parentNestedDataPath= fields.attachment[*] to extract the attachment metadata list from the issue object 2) setting prop parentDataKey=id to extract the 'id' from each attachment metadata. The download request will be performed per each attachment metadata found

Reply:
???

Reply:
@DavidDwyer87 this explanation is to describe how the `/rest/api/2/attachment/content/${LW_PARENT_DATA_KEY}` will download the attachments